CCBoot.js

```diff
@@ -501,6 +501,23 @@ cc.path = /** @lends cc.path# */{
         index = pathStr.lastIndexOf("/");
         index = index <= 0 ? 0 : index + 1;
         return pathStr.substring(0, index) + basename + ext + tempStr;
+    },
+    //todo Open up after verification
+    _normalize: function(url){
+        url = String(url);
+        var status = true;
+        var repFunc = function(){
+            status = true;
+            return "";
+        };
+        //remove ../
+        while(status){
+            status = false;
+            url = url.replace(/[^\.\/]+\/\.\.\//, repFunc);
```

Comment:
I think it's better to replace it directly with an empty string "", and check whether the new url length equals to the old one to determine the status. Because repFunc is likely to make _normalize unacceptable by JIT.

```
var status = true;
var oldUrl;
do {
    oldUrl = url;
    url = oldUrl.replace(/[^\.\/]+\/\.\.\//, "");
}
while (oldUrl.length === url.length);
```